File: Data Structures & Algorithms/jump-game/submission-3.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_index = 0

        for i in range(0,len(nums)):
            if(max_index < i):
                return False

            max_index = max(max_index,nums[i]+i)
            if(max_index >= len(nums)-1):
                return True

        return False            

        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> elements = Arrays.stream(nums).boxed().collect(Collectors.toSet());

        int max = 0, len = 1;

        for(int num : nums) {
            len =1;
            if(!elements.contains(num-1)) {
                while(elements.contains(num+1)) {
                    num++;
                    len++;
                }
            }
            max = Math.max(len,max);
        }

        return max;


    }
}
